src/internet_identity/src/storage.rs

```diff
@@ -597,6 +680,331 @@ impl<M: Memory + Clone> Storage<M> {
+            .get(&(anchor_number, app_num))
+        {
+            None => {
+                // If no list exists for this anchor & application,
+                // Create and insert the default and additional account.
+                // This is because we don't create default accounts explicitly.
+                let additional_account_reference = AccountReference {
+                    account_number: Some(account_number),
+                    last_used: None,
+                };
+                let default_account_reference = AccountReference {
+                    account_number: None,
+                    last_used: None,
+                };
+                self.stable_account_reference_list_memory.insert(
+                    (anchor_number, app_num),
+                    vec![default_account_reference, additional_account_reference].into(),
+                );
+            }
+            Some(existing_storable_list) => {
+                // If the list exists, push the new account and reinsert it to memory
+                let mut refs_vec: Vec<AccountReference> = existing_storable_list.into();
+                refs_vec.push(AccountReference {
+                    account_number: Some(account_number),
+                    last_used: None,
+                });
+                self.stable_account_reference_list_memory
+                    .insert((anchor_number, app_num), refs_vec.into());
+            }
+        }
+
+        // Return the new account
+        Ok(Account {
+            account_number: Some(account_number),
+            anchor_number,
+            origin: origin.clone(),
+            last_used: None,
+            name: Some(params.name),
+        })
+    }
+
+    #[allow(dead_code)]
+    /// Returns a list of account references for a given anchor and application.
+    /// If the application doesn't exist, returns a list with a default account reference.
+    /// If the account references doesn't exist, returns a list with a default account reference.
+    pub fn list_accounts(
+        &self,
+        anchor_number: &AnchorNumber,
+        origin: &FrontendHostname,
+    ) -> Result<Vec<AccountReference>, StorageError> {
+        match self.lookup_application_number_with_origin(origin) {
+            None => Ok(vec![AccountReference {
+                account_number: None,
+                last_used: None,
+            }]),
+            Some(app_num) => match self.lookup_account_references(*anchor_number, app_num) {
+                None => Ok(vec![AccountReference {
+                    account_number: None,
+                    last_used: None,
+                }]),
+                Some(refs) => Ok(refs),
+            },
+        }
+    }
+
+    #[allow(dead_code)]
+    /// Returns the requested account.
+    /// If the account number doesn't esist, returns a default Account.
+    /// If the account number exists but the account doesn't exist, returns None.
+    /// If the account exists, returns it as Account.
```

Comment:
Comment didn't match implementation, see below suggestion:

```suggestion
    /// If the application doesn't exist, returns a default Account.
    /// If the application exists but the account doesn't exist, returns None.
    /// If the account exists, returns it as Account.
```

Reply:
Why do you say it doesn't match? This is the read_account, which checks the account_number. If None, returns a default account. Otherwise, it looks it up in memory.

Reply:
```
/// Line 933-942: If the application doesn't exist, returns a default Account.
/// Line 944: If the application exists but the account doesn't exist, returns None.
/// Line 945-950: If the account exists, returns it as Account.
```